File: CodingPractice/src/main/java/Arrays/twoPointer/minSizeSubArray.java
public class minSizeSubArray {


    public static  void main(String arg[]){

        int[] arr = {2,3,1,4};

        System.out.println(minSize(arr,6));
    }

    public static int minSize(int[] arr, int target){


        if(arr.length == 0){

            return 0;
        }

        int i = 0, j = 0, min = Integer.MAX_VALUE;

        int sum = 0;

        while(i < arr.length){

            sum += arr[i];

            while(sum >= target){

                min = Math.min(min, i - j +1);
                sum -= arr[j];
                j++;
            }
            i++;
        }

        return min == Integer.MAX_VALUE ? Integer.MAX_VALUE: min;
    }
}

File: Solved in Python/LeetCode/arrays/minSizeSubArray.py
class Solutioon(object):
    def minSizesubArray(self, arr, target):
        if len(arr) == 0:
            return 0
        

        i = 0 
        j = 0
        minimum = float('inf')

        sum = 0

        while i < len(arr): 

            sum += arr[i]
            i += 1

            while sum >= target:
                minimum = min(minimum, i - j)
                sum -= arr[j]
                j += 1
      
        if minimum == float('inf'):
            return 0
        else:
            return minimum
    
s= Solutioon()

print(s.minSizesubArray([4, 1, 3, 2], 6))
